app/src/debug/java/org/simple/clinic/ReadOnlySqliteDatabaseDriver.kt

```diff
@@ -0,0 +1,54 @@
+package org.simple.clinic
+
+import android.content.Context
+import android.database.sqlite.SQLiteDatabase
+import com.facebook.flipper.plugins.databases.impl.SqliteDatabaseConnectionProvider
+import com.facebook.flipper.plugins.databases.impl.SqliteDatabaseDriver
+import com.facebook.flipper.plugins.databases.impl.SqliteDatabaseProvider
+import java.io.File
+
+class ReadOnlySqliteDatabaseDriver(
+    context: Context
+) : SqliteDatabaseDriver(
+    context,
+    ContextBasedDatabaseProvider(context),
+    ReadOnlySqliteDatabaseConnectionProvider()
+)
+
+private class ContextBasedDatabaseProvider(private val context: Context) : SqliteDatabaseProvider {
+
+  override fun getDatabaseFiles(): MutableList<File> {
+    return context
+        .databaseList()
+        .map(context::getDatabasePath)
+        .toMutableList()
+  }
+}
+
+private class ReadOnlySqliteDatabaseConnectionProvider : SqliteDatabaseConnectionProvider {
+
+  override fun openDatabase(databaseFile: File): SQLiteDatabase {
+    // We are setting this to READONLY for a couple of reasons:
```

Comment:
Are these three points, three different reasons? 
I think its more like one detailed reason. Maybe consider removing the numbering and just keep the paragraphs.

Reply:
Good point, I'll do this.

Reply:
Done.